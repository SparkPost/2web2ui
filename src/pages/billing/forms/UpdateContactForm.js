import React, { Component } from 'react';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';

import { updateBillingContact, getBillingCountries } from 'src/actions/billing';
import { showAlert } from 'src/actions/globalAlert';
import { updateContactInitialValues } from 'src/selectors/accountBillingForms';

import { Button, Panel } from 'src/components/matchbox';
import { ButtonWrapper } from 'src/components';
import { Form } from 'src/components/tracking/form';
import BillingContactForm from './fields/BillingContactForm';

const FORMNAME = 'updateContact';

export class UpdateContactForm extends Component {
  componentDidMount() {
    this.props.getBillingCountries();
  }

  onSubmit = values => {
    const { updateBillingContact, showAlert } = this.props;
    return updateBillingContact(values).then(() => {
      showAlert({ type: 'success', message: 'Billing Contact Updated' });
    });
  };

  render() {
    const { onCancel, handleSubmit, submitting } = this.props;

    return (
      <Form onSubmit={handleSubmit(this.onSubmit)} id="billing-update-contact">
        <Panel.LEGACY title="Update Billing Contact">
          <Panel.LEGACY.Section>
            <BillingContactForm
              formName={FORMNAME}
              disabled={submitting}
              countries={this.props.billing.countries}
            />
          </Panel.LEGACY.Section>
          <Panel.LEGACY.Section>
            <ButtonWrapper>
              <Button type="submit" variant="primary" disabled={submitting}>
                Update Billing Contact
              </Button>
              <Button onClick={onCancel} variant="secondary">
                Cancel
              </Button>
            </ButtonWrapper>
          </Panel.LEGACY.Section>
        </Panel.LEGACY>
      </Form>
    );
  }
}

const mapStateToProps = state => ({
  billing: state.billing,
  initialValues: updateContactInitialValues(state),
});

const mapDispatchtoProps = { getBillingCountries, updateBillingContact, showAlert };
const formOptions = { form: FORMNAME, enableReinitialize: true };
export default connect(
  mapStateToProps,
  mapDispatchtoProps,
)(reduxForm(formOptions)(UpdateContactForm));
