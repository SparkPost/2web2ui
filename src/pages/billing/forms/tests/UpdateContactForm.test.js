import React from 'react';
import { UpdateContactForm } from '../UpdateContactForm';
import { shallow } from 'enzyme';

describe('Form Container: Update Contact', () => {
  let wrapper;

  const props = {
    billing: { countries: []},
    handleSubmit: jest.fn(),
    showAlert: jest.fn(),
    getBillingCountries: jest.fn(),
    updateBillingContact: jest.fn(() => Promise.resolve()),
    onCancel: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<UpdateContactForm {...props} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should get countries on mount', () => {
    const countriesSpy = jest.spyOn(wrapper.instance().props, 'getBillingCountries');
    wrapper.instance().componentDidMount();
    expect(countriesSpy).toHaveBeenCalled();
  });

  it('should submit redux-Form', () => {
    const submitSpy = jest.spyOn(wrapper.instance().props, 'handleSubmit');
    wrapper.find('Form').simulate('submit');
    expect(submitSpy).toHaveBeenCalled();
  });

  it('should update contact', () => {
    const updateSpy = jest.spyOn(wrapper.instance().props, 'updateBillingContact');
    wrapper.instance().onSubmit('hello');
    expect(updateSpy).toHaveBeenCalledWith('hello');
  });
});
