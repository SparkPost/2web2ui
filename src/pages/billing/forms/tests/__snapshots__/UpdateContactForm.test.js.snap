// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Container: Update Contact should render 1`] = `
<Form
  id="billing-update-contact"
>
  <Panel.LEGACY
    title="Update Billing Contact"
  >
    <Panel.LEGACY.Section>
      <Connect(BillingContactForm)
        countries={Array []}
        formName="updateContact"
      />
    </Panel.LEGACY.Section>
    <Panel.LEGACY.Section>
      <ButtonWrapper>
        <Button
          type="submit"
          variant="primary"
        >
          Update Billing Contact
        </Button>
        <Button
          onClick={[MockFunction]}
          variant="secondary"
        >
          Cancel
        </Button>
      </ButtonWrapper>
    </Panel.LEGACY.Section>
  </Panel.LEGACY>
</Form>
`;
